### PR TITLE
Create echo-tech-spec.md

### PR DESCRIPTION
**Pull request by lomky**
Creates the first draft of the Tech Spec Template.

Based off of Nava's with our team's process attached.

### CODE REVIEW

**Comment by yoomlam** [comment]
Suggest borrowing from the existing templates:
* https://raw.githubusercontent.com/department-of-veterans-affairs/caseflow/master/.github/ISSUE_TEMPLATE/engineering-task.md
* https://raw.githubusercontent.com/department-of-veterans-affairs/caseflow/master/.github/ISSUE_TEMPLATE/epic.md

Specifically, use metadata to set the 'tech-spec' label by default and use html comments to describe each section.

File: .github/ISSUE_TEMPLATE/echo-tech-spec.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+assignees: ''
+
+---
+
+# Tech Spec Title
+**Drafter**: <!-- Your name -->
+**Discussion Meeting**: <!-- Date to discussion the spec as a team; send calendar invite! -->  
+
+<!-- Tech specs are a lightweight format for documenting technical research and decision making. The headings below are guidelines, not rules, so modify your copy as you see fit. Especially large tech specs may require more headings and subheadings for example. Be rigorous in your research and planning, but balance that with forward progress - if something is uncertain, document that appropriately and move on if you can. Getting feedback early from team mates during tech spec writing can be helpful too.
+Tech specs are considered 'done' when key stakeholders have reviewed and approved the approach.
```
Comment (hschallhorn) [inline .github/ISSUE_TEMPLATE/echo-tech-spec.md]: Missing comment closure makes Context header hidden

File: .github/ISSUE_TEMPLATE/echo-tech-spec.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Context
```
Comment (yoomlam) [inline .github/ISSUE_TEMPLATE/echo-tech-spec.md]: `Context` section is being commented out. Need to close the html comment in prior paragraph.

File: .github/ISSUE_TEMPLATE/echo-tech-spec.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+<!-- Who are the stakeholders? -->
+
+## Overview
+<!-- A brief summary of research, findings, and recommendations. -->
+
+## Open Questions
+<!-- Sometimes we're missing information needed to fully spec work. 
+What is missing? Who can answer these questions, and how might it affect the recommendation? -->
+
+## Implementation
```
Comment (hschallhorn) [inline .github/ISSUE_TEMPLATE/echo-tech-spec.md]: Would prefer this to be "Implementation options" to emphasize that multiple approaches should be considered. Comments could be updated to reflect this as well

File: .github/ISSUE_TEMPLATE/echo-tech-spec.md (hunk 1 above)
Comment (lomky) [inline .github/ISSUE_TEMPLATE/echo-tech-spec.md]:
```suggestion
Tech specs are considered 'done' when key stakeholders have reviewed and approved the approach. -->
```

File: .github/ISSUE_TEMPLATE/echo-tech-spec.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
```
Comment (lomky) [inline .github/ISSUE_TEMPLATE/echo-tech-spec.md]:
```suggestion
## Implementation Options
```

File: .github/ISSUE_TEMPLATE/echo-tech-spec.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+<!-- Recommended action items.
```
Comment (lomky) [inline .github/ISSUE_TEMPLATE/echo-tech-spec.md]:
```suggestion
<!-- 
Consider multiple implementation paths.
What are their recommended action items?
```

File: .github/ISSUE_TEMPLATE/echo-tech-spec.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+name: Echo Tech Spec task
+about: Template for a new tech spec for the echo team
+title: '[SUBJECT] Tech Spec'
+labels: 'tech-spec, echo'
+assignees: ''
+
+---
+
+# Tech Spec Title
+**Drafter**: <!-- Your name -->
```
Comment (lomky) [inline .github/ISSUE_TEMPLATE/echo-tech-spec.md]:
```suggestion
**Drafter**: <!-- Your name -->  
```

**Pull request merged**